ui/src/lib/helpers.tsx

```diff
@@ -13,8 +13,12 @@ import { Middleware } from 'swr';
 /**
  * @description Truncates a string
  */
-export const truncateString = (value: string, length: number): string => {
-    return value.length ? (length < value.length ? `${value.substring(0, length)}...` : value) : value;
+export const truncateString = (string: string, length: number): string => {
+    if (length <= 3) {
+        return '...';
+    }
+    const sliceLength = length - 3;
+    return string.length > sliceLength ? string.slice(0, sliceLength) + '...' : string;
```

Comment:
It would be a good idea to trim whitespace from both ends of the string before taking first "n" characters and even after the slice, so that the description doesn't look like this: "some text      
             ..."

Reply:
Added a commit to do this, good idea